src/Net/TcpSettings.cs

```diff
@@ -87,6 +99,7 @@ public int SendTimeout
 
         internal void Configure(Socket socket)
         {
+            if (this.KeepAlive != null) socket.SetTcpKeepAlive(this.KeepAlive.Item1, this.KeepAlive.Item2);
```

Comment:
Return value of SetTcpKeepAlive is ignored, which results in non-deterministic behavior of the API. The user may think the setting is in effect but it actually failed to apply. We should just let the set method throw so user's code can handle the error.